raster/src/main/scala/geotrellis/raster/render/BreakMap.scala

```diff
@@ -0,0 +1,133 @@
+package geotrellis.raster.render
+
+import geotrellis.raster._
+import geotrellis.util._
+
+import spire.algebra._
+import spire.math.Sorting
+import spire.std.any._
+import spire.syntax.order._
+
+import scala.specialized
+
+// --- //
+
+/** Root element in hierarchy for specifying the type of boundary when classifying colors*/
+sealed trait ClassBoundaryType
+case object GreaterThan extends ClassBoundaryType
+case object GreaterThanOrEqualTo extends ClassBoundaryType
+case object LessThan extends ClassBoundaryType
+case object LessThanOrEqualTo extends ClassBoundaryType
+case object Exact extends ClassBoundaryType
+
+/** A strategy for mapping values via a [[BreakMap]].
+  *
+  * '''Note:''' Specialized for `Int` and `Double`.
+  */
+class MapStrategy[@specialized(Int, Double) A](
+  val boundary: ClassBoundaryType,
+  val noDataValue: A,
+  val fallbackValue: A,
+  val strict: Boolean
+)
```

Comment:
`MapStrategy` might deserve its own file, although it's strictly subordinate to `BreakMap`.